File: aisile-manager-web/src/main/webapp/js/service/goodsService.js
//封装所有连接后端代码    需要$http
app.service('goodsService',function($http){
	//查询全部的数据
	this.findAll = function(){
		return $http.get('../goods/findAll.do');
	}
	//分页
	this.findPage = function(page,rows){
		return $http.get('../goods/findPage.do?page='+page+"&rows="+rows);
	}
	//分页带数据查询
	this.findSearch = function(page,rows,searchEntity){
		return $http.post('../goods/findSearch.do?page='+page+"&rows="+rows,searchEntity);
	}
	//修改
	this.update = function(entity){
		return $http.post('../goods/update.do',entity);
	}
	//添加
	this.add = function(entity){
		return $http.post('../goods/add.do',entity);
	}
	//根据id查询对象
	this.findOne = function(id){
		return $http.post('../goods/findOne.do?id='+id);
	}
	//删除
	this.dele = function(selectIds){
		return $http.get('../goods/dele.do?ids='+selectIds);
	}
	//驳回
	this.turnDown = function(selectIds){
		return $http.get('../goods/turnDown.do?ids='+selectIds);
	}
	//审核通过
	this.openGo = function(selectIds){
		return $http.get('../goods/openGo.do?ids='+selectIds);
	}
	//下拉框
	this.selectOptionList = function(){
		return $http.get('../goods/selectOptionList.do');
	}
})

File: aisile-sellergoods-interface/src/main/java/com/aisile/sellergoods/service/GoodsService.java
package com.aisile.sellergoods.service;

import java.util.List;

import com.aisile.pojo.TbGoods;
import com.aisile.pojo.entity.PageResult;
import com.aisile.pojo.entity.group.Goods;

public interface GoodsService {
	/**
	 * 查询全部数据
	 * @return
	 */
	public List<TbGoods> findAll();
	/**
	 * 分页
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	/**
	 * 模糊查询
	 * @param pageNum
	 * @param pageSize
	 * @param TbGoods
	 * @return
	 */
	public PageResult findSearch(int pageNum,int pageSize,TbGoods tbGoods);
	/**
	 * 添加方法
	 * 使用的是组合类
	 * @param Goods
	 */
	public void add(Goods goods);
	/**
	 * 修改方法
	 * @param TbGoods
	 */
	public void update(Goods goods);
	/**
	 * 回显   返回的是一个对象
	 * @param id
	 * @return
	 */
	public Goods findOne(Long id);
	/**
	 * 删除(批量删除)
	 * @param ids
	 */
	public void delete(Long [] ids);
	/**
	 * 驳回
	 * @param ids
	 */
	public void turnDown(Long[] ids);
	/**
	 * 申请通过
	 * @param ids
	 */
	public void openGo(Long[] ids);
	/**
	 * 上下架处理
	 * @param ids
	 * @param status
	 */
	public void upDownFrame(Long [] ids,String status);
}
